app/Actions/Server/ValidateServer.php

```
@@ -45,6 +45,16 @@ public function handle(Server $server)
             throw new \Exception($this->error);
         }
 
+        $validationResult = $server->validatePrerequisites();
+        if (! $validationResult['success']) {
+            $missingCommands = implode(', ', $validationResult['missing']);
+            $this->error = "Prerequisites ({$missingCommands}) are not installed. Please install them before continuing or use the validation with installation endpoint.";
+            $server->update([
+                'validation_logs' => $this->error,
+            ]);
+            throw new \Exception($this->error);
```

PHPMD (2.15.0)

Missing class import via use statement (line '55', column '23'). (undefined)

(MissingImport)